### ISSUE TITLE
Lanes: multiple indications per lane

### ISSUE THREAD

**Issue by daniel-j-h** [Project-OSRM/osrm-frontend-v2#192]
Lanes with multiple indications per lane are broken on the demo server. Example:

![frontend-lanes](https://cloud.githubusercontent.com/assets/527241/19233478/fec14b06-8ee5-11e6-8569-6b46616b2b40.png)

Check icons: I should take the right most lane going straight. But the tagging is `through|through;right`:

http://map.project-osrm.org/?z=18&center=51.044660%2C13.727455&loc=51.044588%2C13.727047&loc=51.044490%2C13.727878&hl=en&alt=0
![frontend-tags](https://cloud.githubusercontent.com/assets/527241/19233480/01792436-8ee6-11e6-88db-4198e79d50d6.png)

https://www.openstreetmap.org/edit#map=20/51.04484/13.72787

Response from `osrm-backend` is okay:

```
http 'http://router.project-osrm.org/route/v1/driving/13.727047,51.044588;13.727878,51.044490?steps=true'
```

``` json
[{"lanes":[{"valid":false,"indications":["straight"]},{"valid":true,"indications":["straight","right"]}]
```

This has to generate `^ ^>` icons instead of `^ ^`.

**Comment by daniel-j-h** [comment]
The frontend is always using the first indication here: https://github.com/Project-OSRM/osrm-frontend/blob/a486a1e268f3e3f31b8f05a66534c5cc7164d6b6/src/itinerary_builder.js#L25

**Comment by daniel-j-h** [comment]
@freenerd if you're back on frontend work, could you prioritize this. I looked into the frontend with @TheMarex last week for a day but we had to fix the setup, uturn lanes and some icon issues first.

**Comment by freenerd** [comment]
I did not immediately tackle this problem, because it's a bit more complicated:
- It's not clear if we'd want to always show all lane indications (e.g. u-turn might be confusing most of the time)
- It's not clear how to do combined lane symbols in the web
  - We can do composites with svg
  - We can have all combinations in the icon map (which will be huge)
- We do not have sufficient information in lanes to make really nice symbols https://github.com/Project-OSRM/osrm-backend/issues/2935

I have made a small fix in https://github.com/Project-OSRM/osrm-frontend/pull/197 that at least avoids super-confusing lane indications, but does not do combinations.

**Comment by daniel-j-h** [comment]
We can not overlay icons since they're not properly aligned in the sprite. There are no svg icons. You can show multiple indications per lane in the instructions by putting a divider between several icon groups

e.g. for `left;through|through|through;right`

```
< ^  |  ^  |  ^ >
```

**Comment by freenerd** [comment]
#197 is deployed.

> We can not overlay icons since they're not properly aligned in the sprite. There are no svg icons.

These are things we should fix.

> You can show multiple indications per lane in the instructions by putting a divider between several icon groups

I think this will be highly confusing to drivers, since #arrows !== #lanes, thus we should avoid it.

**Comment by yuryleb** [comment]
IMHO displaying necessary maneuver on allowed lanes and graying out arrows on all other lanes proposed in #239 makes this issue not so actual. Of course, it's not so good to display only first indication for rest non-allowed lanes but using overlays or multifications is much worse IMHO - this is definitely useless and extra info for the drivers.

**Comment by freenerd** [comment]
@yuryleb I agree that having too much lane information may be confusing and the changes in #239 make this nicer.

Still, we want to support all-turn-information-per-lane eventually, since it should be up to the developer to decide what they want. Right now we don't support it at all, which is bad, thus it makes sense for this ticket to stay open and wait for https://github.com/Project-OSRM/osrm-backend/issues/2935 to be dealt with.

**Comment by yuryleb** [comment]
Implemented with #284
